spec/queries/display_ads/filtered_ads_query_spec.rb

```diff
@@ -38,13 +39,15 @@ def filter_ads(**options)
     let!(:mismatched) { create_display_ad cached_tag_list: "career" }
 
     it "will show no-tag display ads if the article tags do not contain matching tags" do
-      filtered = filter_ads(article_tags: %w[javascript])
+      allow(article).to receive(:cached_tag_list_array).and_return(%w[javascript])
```

Comment:
All the article stubbing feels rather awkward, but I can't really think of a better way to do it

Reply:
An option here could be to pass `article` into `DisplayAd.for_display`, but decompose into attributes that are passed into `FilteredAdsQuery` — these tests could then revert back to dealing with an options hash, but we'd still have callers (eg, views/controller) would be simplified. 

Do you think that would make this any better?

Reply:
@filleduchaos @Ridhwana thoughts on this approach 7c81fbb0 ?

Reply:
Looks good to me! also yay for tiny diffs

Reply:
I preferred the previous approach where all the logic and decomposition was encapsulated in the FilteredAdsQuery, but I do see the benefits in simplifying the tests and don't have a super strong opinion on it, so I'm good with this 👍🏽